File: hrms-backend/src/main/java/com/finalproject/hrmsbackend/business/abstracts/SystemEmployeeService.java
package com.finalproject.hrmsbackend.business.abstracts;

import com.finalproject.hrmsbackend.core.utilities.results.DataResult;
import com.finalproject.hrmsbackend.core.utilities.results.Result;
import com.finalproject.hrmsbackend.entities.concretes.SystemEmployee;
import com.finalproject.hrmsbackend.entities.concretes.dtos.SystemEmployeeAddDto;

import java.util.List;

public interface SystemEmployeeService {

    DataResult<List<SystemEmployee>> getAll();

    DataResult<SystemEmployee> getById(int sysEmplId);

    DataResult<SystemEmployee> getByEmailAndPW(String email, String password);

    Result add(SystemEmployeeAddDto systemEmployeeAddDto);

    Result updateFirstName(String firstName, int sysEmplId);

    Result updateLastName(String lastName, int sysEmplId);

}

File: hrms-frontend/src/services/systemEmployeeService.js
import axios from "axios"

export default class SystemEmployeeService {

    getAll() {
        return axios.get("http://localhost:8080/api/systemEmployees/get/all");
    }

    getById(sysEmplId) {
        return axios.get(`http://localhost:8080/api/systemEmployees/get/byId?sysEmplId=${sysEmplId}`);
    }

    getByEmailAndPassword(email, password) {
        return axios.get(`http://localhost:8080/api/systemEmployees/get/byEmailAndPW?email=${email}&password=${password}`)
    }

    add(values) {
        return axios.post("http://localhost:8080/api/systemEmployees/add", values)
    }

    updateFirstName(sysEmplId, firstName) {
        return axios.put(`http://localhost:8080/api/systemEmployees/update/firstName?firstName=${firstName}&sysEmplId=${sysEmplId}`)
    }

    updateLastName(sysEmplId, lastName) {
        return axios.put(`http://localhost:8080/api/systemEmployees/update/lastName?lastName=${lastName}&sysEmplId=${sysEmplId}`)
    }

}